src/View3d.ts

```diff
@@ -259,10 +270,11 @@ export class View3d {
     this.loadErrorHandler = handler;
   }
 
-  setTime(volume: Volume, time: number, onChannelLoaded?: PerChannelCallback): void {
+  async setTime(volume: Volume, time: number, onChannelLoaded?: PerChannelCallback): Promise<void> {
     const timeClamped = Math.max(0, Math.min(time, volume.imageInfo.times - 1));
-    volume.updateRequiredData({ time: timeClamped }, onChannelLoaded);
+    const loadPromise = volume.updateRequiredData({ time: timeClamped }, onChannelLoaded);
     this.updateTimestepIndicator(volume);
+    return loadPromise;
```

Comment:
Hmm, okay, so, thinking about this more.

@toloudis I think I could get away with not making `setTime` async? I'd have to do some promise juggling at the TFE level but it's possible if we really don't like this.

Reply:
Basically I'd have to create an empty Promise in TFE and save it to the ColorizeCanvas3D state. I'd wait until a callback I pass to `onChannelLoaded` registers that the volume has loaded, and if so I'd resolve that empty Promise.

Reply:
i think in general it's fine if setTime is async.

Reply:
Came up when we discussed this, but the (minor) worry here is if other state setters that can update what data we need (z slice or channels, for instance) return `Promise`s, then clients start depending on those `Promise`s, then multiple settings like that get set in order and earlier `Promise`s start doing unexpected things. Should be safe as long as it's isolated to just this one setter though, even if it breaks the paradigm a tiny bit.